pkg/controllers/deprovisioning/suite_test.go

```diff
@@ -2059,7 +2058,7 @@ var _ = Describe("Parallelization", func() {
 		pods := ExpectProvisionedNoBinding(ctx, env.Client, provisioningController, provisioner, test.UnschedulablePod())
 		nodes := &v1.NodeList{}
 		Expect(env.Client.List(ctx, nodes)).To(Succeed())
-		Expect(len(nodes.Items)).To(Equal(3))
+		Expect(len(nodes.Items)).To(Equal(2))
```

Comment:
This one is a bit baffling to me. Reading the test, it appears to be testing the opposite behavior as the comment suggests: "Add a new pending pod that should schedule while node is not yet deleted". Before the cluster state change, this was causing a fresh node to be provisioned, resulting in 3 instead of 2 nodes.

When the second node is created, it's not tracked by cluster state. Now, the provisioning loop tracks the cluster state and reuses the node for the provisioning loop. @tzneal, does this sound right?

Reply:
Yeah, I'm not sure of what that old test did either.  The behavior you've described sounds correct now though.